cli/azd/cmd/deploy.go

```diff
@@ -119,10 +119,6 @@ func (d *deployAction) Run(ctx context.Context) error {
 		return err
 	}
 
-	if err := ensureLoggedIn(ctx); err != nil {
-		return fmt.Errorf("failed to ensure login: %w", err)
-	}
```

Comment:
This is now handled as part of the arrangement of dependencies handled by wire.go. `newCredential` now takes an auth.Manager and calls CurrentUser on it, which checks to see if the user is logged in and if their credentials are still valid.  If this passes, it returns the Credential back, otherwise returns ErrNotLoggedIn.  Since the credential object is required to build the `AzCli` instance, any command which dependes on the AzCli now gets this checking for free.